aten/src/ATen/cpu/vec/vec256/vec256_bfloat16.h

```diff
@@ -161,13 +161,13 @@ static_assert(
     __m256i cmp = _mm256_cmpeq_epi16(values, _mm256_set1_epi16(0));
     return _mm256_movemask_epi8(cmp);
   }
-  static Vectorized<T> loadu(const void* ptr) {
-    return _mm256_loadu_si256(reinterpret_cast<const __m256i*>(ptr));
-  }
-  static Vectorized<T> loadu(const void* ptr, int16_t count) {
+  static Vectorized<T> loadu(const void* ptr, int16_t count = size()) {
```

Comment:
I am not sure why this will bring perf improvement. To my understanding, the compiler would have done this type of optimization.

Can you provide some benchmarks or compiled assembly for me to understand the change here?

Also I suppose we cannot use masked load even on AVX512 (since don't have masked load for epi16), right ?

Reply:
> To my understanding, the compiler would have done this type of optimization.

On the Inductor side, we always invoke `loadu` with `count` for the C++/OpenMP backend. Take BF16 memory copy as an example on the AVX512 platform, the code snippet is as follows.

```C++
for (size_t i = 0; i < 1024 * 1024 *1024 / 32; i++) {
    auto tmp0 = at::vec<at::BFloat16>::loadu(in_ptr, 32);
    tmp0.store(out_ptr, 32);
}
```

For BF16, the `loadu` w/ `count` always introduces a temporary buffer even if the `count` equals `size()`. This PR intends to optimize it.